dspot/src/main/java/eu/stamp_project/utils/program/InputConfiguration.java

```diff
@@ -346,8 +349,62 @@ public static void setUp(List<String> amplifiers, String budgetizer,
                 .setNumberParallelExecutionProcessors(numberParallelExecutionProcessors);
     }
 
-    /*
-        Paths project properties
+    public static void setUp(List<String> amplifiers, String budgetizer,String mongoUrl,
+                             TestSelector testCriterion, List<String> testClasses,
+                             List<String> testCases, int iteration,
+                             long seed, int timeOut,
+                             int maxTestAmplified, boolean clean,
+                             boolean verbose, boolean workingDirectory,
+                             boolean comment, boolean generateNewTestClass,
+                             boolean keepOriginalTestMethods, boolean gregor,
+                             boolean descartes, boolean useMavenToExecuteTest,
+                             boolean targetOneTestClass, boolean allowPathInAssertion,
+                             boolean executeTestsInParallel, int numberParallelExecutionProcessors,
+                             String collector) {
+        setUp(amplifiers, budgetizer,
+            testCriterion, testClasses,
+            testCases, iteration,
+            seed, timeOut,
+            maxTestAmplified, clean,
+            verbose, workingDirectory,
+            comment, generateNewTestClass,
+            keepOriginalTestMethods, gregor,
+            descartes, useMavenToExecuteTest,
+            targetOneTestClass, allowPathInAssertion,
+            executeTestsInParallel, numberParallelExecutionProcessors);
+        collectorInit(mongoUrl,collector);
+    }
+
+
+    /* DspotInformation collector related */
+    private static DspotInformationCollector collector = new NullCollector();
+
+    public static void collectorInit(String mongoUrl, String collector) {
+        InputConfiguration.get().setMongoUrl(mongoUrl).setInformationCollector(CollectorEnum.valueOf(collector).getCollector());
```

Comment:
Hello @Tailp 
here, the call to `setMongoUrl` makes this statement mongo specific. Could we have a global API shared by all the `Collector`s? Do we need to keep the mongoUrl inside the `InputConfiguration` or could we put it inside the `MongoCollector`?

Reply:
Hi @danglotb :) , I used InputConfiguration as a global config for storing my collector parsed JSAP values , so I can just access those with a getter later . This also make things easier later on for different kinds of collector since it can just fetch the stored JSAP values from inputConfiguration upon instantiation without any initialization :).

Reply:
Yes, I agree with you, However, I'm asking myself: Is anywhere else than in the `mongoCollector` where you use the `mongoUrl`? If no, It seems legit to put this url inside the `mongoCollector`, right? As field for instance, WDYT?

Reply:
@danglotb sure :) , but I might have to add an initialize method to the interface as a result to keep the polymorphism Informationcollector.

Reply:
Hi @danglotb :) , I removed all collector related in InputConfig and instead added a new CollectorConfig class to be configured by Configuration.java and store information from JSAP . I can use this class to fetch information for each collector kind upon instatiation . So the base idea is the same , but this make the code a bit shorter since we skipped InputInformationCollector.